src/Compilers/CSharp/Portable/Symbols/Metadata/PE/PENamedTypeSymbol.cs

```diff
@@ -487,22 +478,26 @@ private ImmutableArray<NamedTypeSymbol> MakeDeclaredInterfaces()
 
                 if (interfaceImpls.Count > 0)
                 {
-                    NamedTypeSymbol[] symbols = new NamedTypeSymbol[interfaceImpls.Count];
+                    var symbols = ArrayBuilder<NamedTypeSymbol>.GetInstance(interfaceImpls.Count);
                     var tokenDecoder = new MetadataDecoder(moduleSymbol, this);
 
-                    int i = 0;
                     foreach (var interfaceImpl in interfaceImpls)
                     {
                         EntityHandle interfaceHandle = moduleSymbol.Module.MetadataReader.GetInterfaceImplementation(interfaceImpl).Interface;
                         TypeSymbol typeSymbol = tokenDecoder.GetTypeOfToken(interfaceHandle);
-
-                        typeSymbol = TupleTypeDecoder.DecodeTupleTypesIfApplicable(typeSymbol, interfaceImpl, moduleSymbol);
-
-                        var namedTypeSymbol = typeSymbol as NamedTypeSymbol;
-                        symbols[i++] = (object)namedTypeSymbol != null ? namedTypeSymbol : new UnsupportedMetadataTypeSymbol(); // interface tmpList contains a bad type
+                        var result = typeSymbol as NamedTypeSymbol;
+                        if (result is null)
+                        {
+                            result = new UnsupportedMetadataTypeSymbol(); // interface tmpList contains a bad type
```

Comment:
>UnsupportedMetadataTypeSymbol [](start = 41, length = 29)

Is there a way to provide some details in the `UnsupportedMetadataTypeSymbol`? That would probably help for troubleshooting any problems that arise.

Reply:
Perhaps, although we should make such a change in `master`.

---
In reply to: [194198601](https://github.com/dotnet/roslyn/pull/27589#discussion_r194198601) [](ancestors = 194198601)